File: 0011. Container With Most Water/Solution.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        res = 0
        while l < r:
            if height[l] < height[r]:
                res = max(res, (r - l) * height[l])
                l += 1
            else:
                res = max(res, (r - l) * height[r])
                r -= 1
        return res

File: 0011. Container With Most Water/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int res = 0;
        while (left < right) {
            if (height[left] <= height[right]) {
                res = Math.max(res, (right - left) * height[left]);
                left += 1;
            } else {
                res = Math.max(res, (right - left) * height[right]);
                right -= 1;
            }
        }
        return res;
    }
}
